### PR TITLE
add warning and document --dev flag

### PR DESCRIPTION
**Pull request by smikes**
add warning re `npm update -g`
document --dev flag

Fix for #6745.  A mild warning about `npm update -g` was added, as well.

### CODE REVIEW

File: doc/cli/npm-update.md [hunk 1]
```diff
@@ -18,6 +20,15 @@ packages.
 If no package name is specified, all packages in the specified location (global
 or local) will be updated.
 
+## WARNING
```
Comment (othiym23) [inline doc/cli/npm-update.md]: Split this out into a separate patch, plz -- at some point we're going to need to update the docs again to fix this, and it's probably going to be forgotten unless we have some way of tracking that on, say, #6247.

**Comment by othiym23** [comment]
Split out the `npm -g update` warning into a separate PR and this is good to go. Thanks!

**Comment by smikes** [comment]
will do.

**Comment by othiym23** [comment]
Reworded the commit message and squashed into one commit, and then landed as 560c00945d4dec926cd29193e336f137c7f3f951. Once more, thanks!
